### PR TITLE
V4 dev updates and backports

### PR DESCRIPTION
**Pull request by XhmikosR**
TODO:

- [ ] Simplify/improve the newly added scrollspy test cases
- [ ] Tweak the newly added validation workaround examples if needed
- [ ] Do we need to port the workaround to master too?
- [ ] Review the wording changes
- [ ] **Make a final rebase before merge**

Preview:

* <https://deploy-preview-30768--twbs-bootstrap.netlify.app/>
* <https://deploy-preview-30768--twbs-bootstrap.netlify.app/docs/4.4/components/forms/#input-group-validation-workaround>

### CODE REVIEW

**Review by mdo — APPROVED** [review]

File: site/docs/4.4/components/forms.md [hunk 1]
```diff
@@ -978,7 +978,7 @@ Validation styles are available for the following form controls and components:
 
 {% capture callout %}
 ##### Input group validations
-Input groups have difficulty with validation styles unfortunately. Our recommendation is to place feedback messages as sibling elements of the `.input-group` that has `.is-{valid|invalid}`. Placing feedback messages within input groups breaks the `border-radius`. [See this comment for workaround.](https://github.com/twbs/bootstrap/issues/25110#issuecomment-586565165)
+Input groups have difficulty with validation styles, unfortunately. Our recommendation is to place feedback messages as sibling elements of the `.input-group` that has `.is-{valid|invalid}`. Placing feedback messages within input groups breaks the `border-radius`. [See this comment for a workaround](https://github.com/twbs/bootstrap/issues/25110#issuecomment-586565165).
```
Comment (XhmikosR) [inline site/docs/4.4/components/forms.md:981]: @MartijnCuppens @mdo: I'm still not happy with us linking to a GitHub comment. We should move the workaround here.

File: js/src/scrollspy.js [hunk 2]
```diff
@@ -157,11 +157,12 @@ class ScrollSpy {
       ...typeof config === 'object' && config ? config : {}
     }
 
-    if (typeof config.target !== 'string' && Util.isElement(config.target)) {
-      let id = $(config.target).attr('id')
+    if (typeof config.target !== 'string') {
+      const $target = $(document).find(config.target)
```
Comment (Johann-S) [inline js/src/scrollspy.js:161]: what's the point of finding an element already found ?

File: site/index.html [hunk 3]
```diff
@@ -11,7 +11,7 @@
       <div class="col-md-8 order-md-1 text-center text-md-left pr-md-5">
         <h1 class="mb-3">Build fast, responsive sites with Bootstrap</h1>
         <p class="lead mb-4">
-          Quickly design and customize responsive mobile-first sites with Bootstrap, the world’s most popular front-end open source toolkit, featuring Sass variables and mixins, responsive grid system, extensive prebuilt components, and powerful JavaScript plugins.
+          Quickly design and customize responsive mobile-first sites with Bootstrap, the world’s most popular front-end open-source toolkit, featuring Sass variables and mixins, responsive grid system, extensive prebuilt components, and powerful JavaScript plugins.
```
Comment (mdo) [inline site/index.html:14]: Remove the dash

```suggestion
          Quickly design and customize responsive mobile-first sites with Bootstrap, the world’s most popular front-end open source toolkit, featuring Sass variables and mixins, responsive grid system, extensive prebuilt components, and powerful JavaScript plugins.
```

File: site/index.html [hunk 4]
```diff
@@ -85,10 +85,10 @@ <h2>BootstrapCDN</h2>
         </div>
         <h2>Bootstrap Icons</h2>
         <p class="lead font-weight-normal">
-          For the first time ever, Bootstrap has its own open source SVG icon library, designed to work best with our components and documentation.
+          For the first time ever, Bootstrap has its own open-source SVG icon library, designed to work best with our components and documentation.
```
Comment (mdo) [inline site/index.html:88]: Remove the dash here, too.

```suggestion
          For the first time ever, Bootstrap has its own open source SVG icon library, designed to work best with our components and documentation.
```

File: site/docs/4.4/components/forms.md [hunk 5]
```diff
@@ -1153,6 +1153,68 @@ $form-validation-states: map-merge(
 }
 {% endhighlight %}
 
+### Input group validation workaround
+
+When you are using a medium size input group and don't customize the default border radius values, add `.rounded-right` to the missing border radius elements:
```
Comment (mdo) [inline site/docs/4.4/components/forms.md:1158]:
```suggestion
We're unable to resolve the broken `border-radius` of input groups with validation due to selector limitations, so manual overrides are required. When you're using a standard input group and don't customize the default border radius values, add `.rounded-right` to the elements with the broken `border-radius`.
```

File: site/docs/4.4/components/forms.md [hunk 6]
```diff
@@ -1153,6 +1153,68 @@ $form-validation-states: map-merge(
...
+      <span class="input-group-text">@</span>
+    </div>
+    <input type="text" class="form-control is-invalid rounded-right" required>
+    <div class="invalid-feedback">
+      Please choose a username.
+    </div>
+  </div>
+</div>
+
+When you are using a small or large input-group or customizing the default border radius values, add custom CSS to the missing border radius elements:
```
Comment (mdo) [inline site/docs/4.4/components/forms.md:1184]:
```suggestion
When you are using a small or large input group or customizing the default `border-radius` values, add custom CSS to the element with the busted `border-radius`.
```

**Pull request merged**